src/no-theater-available.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  LitElement,
+  customElement,
+  property,
+  html,
+  TemplateResult,
+  PropertyValues,
+  CSSResult,
+  css,
+} from 'lit-element';
+
+import { IntLoadingStateUpdatedEvent } from './interfaces/event-interfaces';
+
+@customElement('ia-no-theater-available')
+export class IANoTheaterAvailable extends LitElement {
+  @property({ type: String }) identifier?: string = '';
+
+  emitLoaded(): void {
+    this.dispatchEvent(
+      new CustomEvent('loadingStateUpdated', {
+        detail: { loaded: true },
+      }) as IntLoadingStateUpdatedEvent
```

Comment:
I'm not sure you need this `as IntLoadingStateUpdatedEvent`. From the dispatcher perspective, it knows what kind of event it is. I think you want this in the consumer of this event so it knows what the `detail` should look like.

I'm not sure if this syntax does the same thing that you're trying to do, but since `CustomEvent` is already a generic, you can do this:
```
new CustomEvent<{ loaded: boolean }>('loadingStateUpdated', { ... });
```

Reply:
i like this more generic syntax better. will update